firebase-ml-modeldownloader/src/main/java/com/google/firebase/ml/modeldownloader/internal/ModelFileManager.java

```diff
@@ -118,6 +126,21 @@ private int getLatestCachedModelVersion(@NonNull File modelDir) {
   @Nullable
   File getModelFileDestination(@NonNull CustomModel model) throws Exception {
     File destFolder = getDirImpl(model.getName());
+    if (!destFolder.exists()) {
+      Log.d(TAG, "model folder does not exist, creating one: " + destFolder.getAbsolutePath());
+      if (!destFolder.mkdirs()) {
+        System.out.println("Failed to create model folder" + destFolder);
+        //        throw new FirebaseMLException(
```

Comment:
Intentional?

Reply:
yes - the logging which will be the commented out lines is currently being tested. Almost there.

Reply:
Sure. Ping me when they're done so I can do a last review before granting approval. Thanks!